lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -225,6 +226,12 @@ void HWMemSimImpl::generateMemory(HWModuleOp op, FirMemory mem) {
   sv::RegOp reg = b.create<sv::RegOp>(
       UnpackedArrayType::get(dataType, mem.depth), b.getStringAttr("Memory"));
 
+  if (!disableMemRandomization) {
+    auto regName = b.getStringAttr(moduleNamespace.newName("_GARBAGE_READ"));
+    b.create<sv::IfDefOp>("RANDOMIZE_GARBAGE_ASSIGN", [&]() {
+      garbageReadRegister = b.create<sv::RegOp>(dataType, regName, regName);
+    });
+  }
```

Comment:
nit:I think this creates one `garbageReadRegister` which is shared by multiple read ports. Is it necessary to create different random values per read port?

Reply:
I think we should create different random values per read port. Updated the PR and also separated out the initialization into its own ifdef instead of `RANDOM_REG_INIT`, which seems incorrect.